ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+        description="What strategy to follow when there's a missing value in a set column",
+    )
+
+    fill_value: Optional[Any] = schema_utils.String(
+        default="<UNK>",
+        allow_none=False,
+        description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+    )
+
+    computed_fill_value: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="The computed fill value determined by the user or inferred from the data.",
+    )
+
+    lowercase: Optional[bool] = schema_utils.Boolean(
+        default=False,
+        description="If true, converts the string to lowercase before tokenizing.",
+    )
+
+    most_common: Optional[int] = schema_utils.PositiveInteger(
+        default=10000,
+        allow_none=True,
+        description="The maximum number of most common tokens to be considered. If the data contains more than this "
+        "amount, the most infrequent tokens will be treated as unknown.",
+    )
+
+
+@dataclass
+@register_preprocessor("h3")
+class H3PreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in an h3 column",
+    )
+
+    fill_value: Optional[Any] = schema_utils.PositiveInteger(
+        default=576495936675512319,
+        allow_none=False,
+        description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+    )
+
+    computed_fill_value: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="The computed fill value determined by the user or inferred from the data.",
+    )
+
+
+@dataclass
+@register_preprocessor("date")
+class DatePreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in a date column",
+    )
+
+    fill_value: Optional[Any] = schema_utils.String(
+        default="",
+        allow_none=False,
+        description="The value to replace missing values with in case the missing_value_strategy is fill_with_const",
+    )
+
+    computed_fill_value: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="The computed fill value determined by the user or inferred from the data.",
+    )
+
+    datetime_format: Optional[str] = schema_utils.String(
+        default=None,
```

Comment:
Hey @connor-mccorm  , this caused a bit of weirdness on the Client. The solution that @hungcs and I came up with for now is to make this field not required. Since Date Time has a standard ISO format, I'm curious if we could come up with a reasonable default that both Python/Ludwig and JavaScript's `Date()` library can interpret. Should we try to pair on that?

Reply:
Not sure about what the default here should be, but to have something better than an unvalidated string...

@connor-mccorm what we could do here is define a custom marshmallow field specifically for datetime. And we would override `_json_type_mapping` as we do in the other complex fields, and in particular set the [`pattern`](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html) attribute to this regex pattern: `^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$` (ref: https://www.myintervals.com/blog/2009/05/20/iso-8601-date-validation-that-doesnt-suck/). Don't necessarily have to do this in this PR though.

@martindavis both RJSF and `Date` should interpret this pattern correctly.

Reply:
Ya I think that a pattern for datetime format is a good call. I will add this in.

Reply:
Talked with @martindavis and decided that a pattern validation for another pattern validation isn't the best idea. Additionally, since the there's not an undeniably most common datetime format, without format inference, it is best to keep the default as None and just require the field when a datetime feature is selected. A future fix for this would be implementing datetime format inference for datetime format features.